content/documentation/architecture.adoc

```diff
@@ -0,0 +1,116 @@
+---
+title: "Architecture"
+date: 2018-09-03T14:39:48-05:00
+draft: false
+---
+
+= Kiali architecture
+:imagesdir: /images/documentation/architecture
+
+Kiali is composed of two components: a backend running in the container
+application platform, and a user frontend. However, in order to be useful,
+Kiali depends on external services and components provided by other
+applications.
+
+The following diagram illustrates the components involved in Kiali and its
+interactions:
+
+image::architecture.png[Kiali architecture]
+
+== Kiali backend
+
+Tha backend is the application that runs in the container application platform.
+It's written in http://golang.org/[Go]. The code can be found at
+https://github.com/kiali/kiali[kiali/kiali GitHub repository].
+
+This is the component that communicates with Istio parts, retrieves and
+processes data, and send processed data to the frontend.
+
+The backend doesn't need storage. When running the backend locally, all
+configurations are set in files and environment variables (including the user
+and password used for authentication). When deploying the backend to a cluster
+using the docker images, configmaps and secrets are used to set the
+configurations.
+
+== Kiali frontend
+
+The frontend is the application that runs in the browser.  It's a React
+application, written in Typescript. The code can be found at
+https://github.com/kiali/kiali-ui[kiali/kiali-ui GitHub repository].
+
+In a standard deployment, the backend provides the frontend, but once the
+browser has downloaded the frontend, it will run mainly in the browser and only
+assets will be requested. The frontend queries the Kiali backend in order to
+get data and present it to the user. 
+
+Since, currently, there are no options for personalizations, it's mainly
+stateless. Some data may be persisted - like session credentials -, but this
+data is stored in the browser and won't be available in other browsers nor
+other devices.
+
+== Istio
+
+Istio is a Kiali requirement. It's the component that provides and controls the
+service mesh. Although Kiali and Istio can be installed separately, Kiali
+depends on Istio and won't be useful if it's not present.
+
+Kiali doesn't communicate with Istio. This is the reason the diagram shows a
+dashed line. Communication between Kiali and Istio occurs through the cluster
+API.
```

Comment:
is there any link to the cluster API documentation?

Reply:
and perhaps it would be beneficial to explain why it does not communicate directly with Istio

Reply:
I don't think there's any real intention in not having any direct communication with Istio. It's just that we have everything we need via prometheus and kubernetes apis.
That could change in the future; ex: with this feature request https://github.com/kiali/kiali/issues/400

Reply:
I changed the paragraph using @jotak comment. Probably now, it better explains the idea.
I added links to API docs in the "Cluster API" section.